### ISSUE TITLE
Service profile autogeneration

### ISSUE THREAD

**Issue by StupidScience** [linkerd/linkerd2#3604]
## Feature Request

Generate service profiles automatically based on Swagger (OpenAPI) specification.

### What problem are you trying to solve?

At the moment you should generate service profiles explicitly. But services usually serve Swagger Spec as well so there is the way to generate Service Profiles based on it.

### How should the problem be solved?

Add kinda annotation with swagger spec path that will be fetched by linkerd control-plane for service profile generation.
I believe here can be some problems in generating new Service Profile at the deployment time.

### Any alternatives you've considered?

Generate it automatically based on `tap` results.

### How would users interact with this feature?

I believe it should be just one more linkerd annotation, e.g. `linkerd.io/service-profile-openapi-spec: /swagger/doc.json`

**Comment by grampelberg** [comment]
Have you seen `linkerd profile --open-api`?

**Comment by StupidScience** [comment]
Yes, ofc. But it should be invoked explicitly.

**Comment by wmorgan** [comment]
@StupidScience To make sure I understand this: your proposal is that users annotate a Deployment object with the URL path of a Swagger-serving endpoint on that Deployment. Then Linkerd calls that endpoint and creates a service profile for the Deployment, and keep it up to date when the contents of that endpoint changes.

This would be an alternative to using `linkerd profile --open-api` and re-applying the service profiles whenever it changes. 

Is that right?

**Comment by StupidScience** [comment]
@wmorgan yep, looks correct.

**Comment by grampelberg** [comment]
Makes sense! This would need to be an annotation on the service and not the deployment as deployments don't have service profiles. Otherwise, sounds like a cool feature! Could be used for protobufs as well.
